src/modules/out_net.c

```diff
@@ -0,0 +1,308 @@
+	// sock:/PATH/PATH/NAME.SOCK,wwxhh,snake
+	if ((ip_sock_str = strsep(&param_str, nettype == NET_SOCK ? "," : ":")) == NULL) {
+		eprintf(PARAM_ERR_STR, "doesn't contain a socket/ip separator");
+		return 3;
+	}
+
+	if (nettype != NET_SOCK) {
+		// tcp:aaa.aaa.aaa.aaa:PPPP,wwxhh,snake
+		// udp:aaa.aaa.aaa.aaa:PPPP,wwxhh,snake
+		if ((port_str = strsep(&param_str, ",")) == NULL) {
+			eprintf(PARAM_ERR_STR, "doesn't contain port");
+			return 3;
+		}
+	} else {
+		port_str = "";
+	}
+
+	// tcp:aaa.aaa.aaa.aaa:pppp,WWxhh,snake
+	// udp:aaa.aaa.aaa.aaa:pppp,WWxhh,snake
+	// sock:/path/path/name.sock,WWxhh,snake
+	if ((x_str = strsep(&param_str, "x")) == NULL) {
+		eprintf(PARAM_ERR_STR, "doesn't contain X size");
+		return 3;
+	}
+
+	X_SIZE = util_parse_int(x_str);
+	if (X_SIZE == 0) {
+		eprintf(PARAM_ERR_STR, "doesn't contain a X matrix size");
+		return 4;
+	}
+
+	// tcp:aaa.aaa.aaa.aaa:pppp,wwxHH,snake
+	// udp:aaa.aaa.aaa.aaa:pppp,wwxHH,snake
+	// sock:/path/path/name.sock,wwxHH,snake
+	if ((y_str = strsep(&param_str, ",")) == NULL) {
+		eprintf(PARAM_ERR_STR, "doesn't contain Y size");
+		return 3;
+	}
+
+	Y_SIZE = util_parse_int(y_str);
+	if (Y_SIZE == 0) {
+		eprintf(PARAM_ERR_STR, "doesn't contain a Y matrix size");
+		return 4;
+	}
+
+	// tcp:aaa.aaa.aaa.aaa:pppp,wwxhh,SNAKE
+	// udp:aaa.aaa.aaa.aaa:pppp,wwxhh,SNAKE
+	// sock:/path/path/name.sock,wwxhh,SNAKE
+	tile_type_str = param_str;
+	tiletype = -1;
+	if (strcmp(tile_type_str, "plain") == 0) tiletype = TILE_PLAIN;
+	if (strcmp(tile_type_str, "snake") == 0) tiletype = TILE_SNAKE;
+	if (tiletype == -1) {
+		eprintf(PARAM_ERR_STR, "doesn't contain a valid tiling type");
+		return 4;
+	}
+
+	// Create socket
+	switch (nettype) {
+		case NET_TCP:
+			sock=socket(AF_INET, SOCK_STREAM, 0);
+			break;
+		case NET_UDP:
+			sock=socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+			break;
+		case NET_SOCK:
+			sock=socket(AF_UNIX, SOCK_STREAM, 0);
+			break;
+	}
+
+	if (sock == -1) {
+		perror("out_net: Failed to initialize socket");
+		return 2;
+	}
+
+	// TCP and UDP related socket initialization
+	if (nettype != NET_SOCK) {
+		memset((char *) &sio_inet, 0, sizeof(struct sockaddr_in));
+		sio_inet.sin_family = AF_INET;
+
+		// Parse IP address
+		if (inet_aton(ip_sock_str, &sio_inet.sin_addr) == 0) {
+			eprintf(PARAM_ERR_STR, "doesn't contain a valid IP");
+			return 3;
+		}
+
+		// Parse port number
+		port = util_parse_int(port_str);
+		if (port == 0) {
+			eprintf(PARAM_ERR_STR, "doesn't contain a valid port number");
+			return 3;
+		}
+		sio_inet.sin_port = htons(port);
+
+		// Connecting to the server socket
+		if (connect(sock, (struct sockaddr*)&sio_inet, sizeof(sio_inet)) != 0) {
+			eprintf("out_net failed to connect to server.");
+			return 5;
+		}
+
+		// This migt be a necessary hack if the OS decides to buffer data
+		// I think this might only be needed for TCP connections... ~esden
+		//int flag = 1;
+		//setsockopt(sock, IPPROTO_TCP, TCP_NODELAY, (char *) &flag, sizeof(int));
```

Comment:
Yes, this is only relevant for TCP. Leaving it out might result in weird behaviour, exactly if the OS decides that it should wait before sending it out, thus not sending the frames frame-by-frame but possibly in any way it deems to be OK.

Reply:
Ok, I have decided to enable this for all TCP connections. I also updated the comment with a warning that this can sometimes be ignored by the OS too.